### PR TITLE
Coach fitz debug

### PR DESCRIPTION
**Pull request by coachFitz0807**
It's probably time to merge this to InfiniteSimulator. I merged fix_startup_jump and the new fieldmap with the shield generator to CoachFitzDebug.

### CODE REVIEW

**Review by LQ1234 — APPROVED** [review]
Looks good to me

**Comment by prensing** [comment]
@coachFitz0807 The 3 commands all do "driveTrain.setPose(initialPose)" at the start in their constructor. However, I think that is redundant. The Robot will already have done that many times in Robot.disabledPeriodic().  Do you think that is needed in the commands?

**Comment by prensing** [comment]
Actually, yes, it is in the constructor of each, so actually is a bad idea. As each command is constructed while building the SendableChooser, the robot will jump around the field (unnecessarily).

File: src/main/java/frc/robot/commands/NewEightBallSim.java [hunk 1]
```diff
@@ -20,9 +20,18 @@
...
+
+    // Define the initial pose to be used by this command. This will be used in the initial trajectory
+    // and will allow the system to query for it
+    private Pose2d initialPose = FieldMap.startPosition[1];
+
     public NewEightBallSim(DriveTrain robotDrive, DriveCommand drivecommand, Climber climber) {
         drivecommand.cancel();
+
+        // Since this is an autonomous command, we need to reset the robot position to the initialPose
+        robotDrive.setPose(initialPose);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/NewEightBallSim.java]: I think this should be removed

File: src/main/java/frc/robot/commands/ShedTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+       // Define the initial pose to be used by this command. This will be used in the initial trajectory
+    // and will allow the system to query for it
+    private Pose2d initialPose = new Pose2d(FieldMap.ballPosition[7], Rotation2d.fromDegrees(0));
+
+    public ShedTest(DriveTrain robotDrive) {
+        // Save the passed in initialPose so we can use it later
+
+
+        // Since this is an autonomous command, we need to reset the robot position to the initialPose
+        robotDrive.setPose(initialPose);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/ShedTest.java]: This should be removed.

File: src/main/java/frc/robot/commands/TrenchAuto.java [hunk 3]
```diff
@@ -20,9 +20,21 @@
...
+    private Pose2d initialPose;
+
+    public TrenchAuto(Pose2d initialPose, DriveTrain robotDrive, DriveCommand drivecommand, Climber climber) {
         drivecommand.cancel();
+
+        // Save the passed in initialPose so we can use it later
+        this.initialPose = initialPose;
+
+        // Since this is an autonomous command, we need to reset the robot position to the initialPose
+        robotDrive.setPose(initialPose);
```
Comment (prensing) [inline src/main/java/frc/robot/commands/TrenchAuto.java]: This should be removed.

**Review by prensing — CHANGES_REQUESTED** [review]

**Review by YJ-7274 — APPROVED** [review]
Everything looks good besides the changes that coach Rensing brought up.

File: src/main/java/frc/robot/commands/NewEightBallSim.java (hunk 1 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/NewEightBallSim.java]: I concur.

File: src/main/java/frc/robot/commands/ShedTest.java (hunk 2 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/ShedTest.java]: I concur

File: src/main/java/frc/robot/commands/TrenchAuto.java (hunk 3 above)
Comment (coachFitz0807) [reply src/main/java/frc/robot/commands/TrenchAuto.java]: I concur.

**Review by coachFitz0807 — COMMENTED** [review]
Implemented changes suggested by Coach Rensing.

**Review by prensing — APPROVED** [review]

**Pull request merged**